### PR TITLE
Prevent unauthenticated account management, list collaborators

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-2171)

## user with collaborators ##

image

## user without collaborators ##

image

### CODE REVIEW

File: company/views.py [hunk 1]
```diff
@@ -563,19 +578,26 @@ def get_success_url(self):
         return settings.SSO_PROFILE_URL + '?user-added'
 
 
-class RemoveCollaboratorView(BaseMultiUserAccountView, FormView):
+class RemoveCollaboratorView(BaseMultiUserAccountManagementView, FormView):
     form_class = forms.RemoveCollaboratorForm
     template_name = 'company-remove-collaborator.html'
 
+    def get_form_kwargs(self, *args, **kwargs):
```
Comment (kowalcj0) [inline company/views.py]: @richtier where is this func used?

File: company/views.py [hunk 2]
```diff
@@ -563,19 +578,26 @@ def get_success_url(self):
...
+        return {
+            'sso_session_id': self.request.sso_user.session_id,
+            **form_kwargs
+        }
+
     def form_valid(self, form):
-        supplier_ids = form.cleaned_data['supplier_ids']
-        self.remove_collaborator(supplier_ids=supplier_ids)
+        sso_ids = form.cleaned_data['sso_ids']
+        self.remove_collaborator(sso_ids=sso_ids)
```
Comment (kowalcj0) [inline company/views.py]: @richtier should we verify whether these sso_ids are actually associated with the account before removing them?

File: company/views.py (hunk 1 above)
Comment (richtier) [reply company/views.py]: [it's used by the framework](https://docs.djangoproject.com/en/1.9/ref/class-based-views/mixins-editing/#django.views.generic.edit.FormMixin.get_form_kwargs) when instantiation `form_class`

File: company/views.py (hunk 2 above)
Comment (richtier) [reply company/views.py]: - The field choices that populate `sso_ids` are provided by the API, so the browser will show only suppliers that are associated with the company.
- If a bad actor manually submits the form, passing in random sso_ids, the API will remove only sso_ids that are related to the current user's company.

So if someone passes in "bad" sso_ids, the action will be idempotent.

**Pull request merged**